server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -93,6 +93,30 @@ public static boolean isValidLongitude(double longitude) {
         return true;
     }
 
+    /**
+     * If the bounding box crosses the date-line (left greater-than right) then the
+     * longitude of the point need only to be higher than the left or lower
+     * than the right. Otherwise, it must be both.
+     *
+     * @param lon the longitude of the point
+     * @param lat the latitude of the point
+     * @param top the northern latitude of the bounding box
+     * @param bottom the southern latitude of the bounding box
+     * @param left the western longitude of the bounding box
+     * @param right the eastern longitude of the bounding box
+     * @return whether the point (lon, lat) is in the specified bounding box
+     */
+    public static boolean pointInBounds(double lon, double lat, double top, double bottom, double left, double right) {
```

Comment:
Can we use Lucene's [Rectangle#containsPoint](https://github.com/apache/lucene-solr/blob/2feeb88c294030be654183d22a8e503bbbf8b293/lucene/core/src/java/org/apache/lucene/geo/Rectangle.java#L88) static method instead and remove this?

Reply:
it didn't look like Rectangle#containsPoint works for Rectangles that cross the dateline? or I might be mis-using it, but I would expect this test to pass

```
    public void testDatelineCrossingRectangleContainsPoint() {
        assertTrue(Rectangle.containsPoint(67.35, 178.41, 175.07, -174.95, 68.57, 66.19));
    }
```